File: java/Gate_five/Testforarray.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;


public class Testforarray{

@Test

    public void testcountEven(){
    
        forarray testing = new forarray();
        int[]arr1actual = testing.countEven(1,2,3,4,5,6);
        int expected = [3];
        assertEquals(actual,expected); 
    }

@Test

    public void testlinearsearch(){
    
        linearsearch testing = new linearsearch();
        int[]arr2actual = testing.linearsearch{10,20,30,40,}
        int[]arr2 expected = [2,-1];
        assertEquals(actual,expected); 
    }


  
@Test

    public void testSquare(){
    
        Square testing = new Square();
        int[] Square actual = testing.Square {5}
        int[] expected = [25];
        assertEquals(actual,expected); 
    }

@Test

    public void testarrayequal(){
    
        arrayEqualtesting = new arrayEqual();
        int[]arr2actual = testing.arrayEqual{1,2,3};
        int[]arr2 expected = [true];
        assertEquals(actual,expected);

File: python/Gate-five/Test_for_array.py
def test_counteven():
    assert counteven([1, 2, 3, 4, 6]) == 3
    assert counteven([1, 3, 5]) == 0
    assert counteven([2, 4, 6, 8]) == 4
    print("counteven passed")

def test_linearsearch():
    assert linearsearch([5, 3, 7, 1], 7) == 2
    assert linearsearch([1, 2, 3], 4) == -1
    assert linearsearch([], 10) == -1
    print("linearsearch passed")

def test_square():
    assert square(5) == [1, 4, 9, 16, 25]
    assert square(1) == [1]
    assert square(0) == []
    print("square passed")

def test_array_equal():
    assert array_equal([1, 2], [1, 2]) == 2
    assert array_equal([1, 2], [2, 1]) == 0
    assert array_equal([1, 2, 3], [1, 2]) == 0
    print("array_equal passed")

def test_remove_duplicate():
    assert remove_duplicate([1, 2, 2, 3, 1]) == [1, 2, 3]
    assert remove_duplicate([5, 5, 5]) == [5]
    assert remove_duplicate([]) == []
    print("remove_duplicate passed")
